### PR TITLE
Add a link to the global settings in the room notifications settings

### PR DESCRIPTION
**Pull request by nimau**
This PR implements #1045 

On the room notification screen, when the user taps on "You can change it in your global settings", the notification settings screen opens modally.

### CODE REVIEW

File: ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift [hunk 1]
```diff
@@ -39,11 +46,16 @@ final class NotificationSettingsScreenCoordinator: CoordinatorProtocol {
...
-                                                        notificationSettingsProxy: parameters.notificationSettings)
+                                                        notificationSettingsProxy: parameters.notificationSettings,
+                                                        isModallyPresented: parameters.isModallyPresented)
     }
     
     func start() {
         viewModel.fetchInitialContent()
+        
+        viewModel.actions.sink { [weak self] _ in
+            self?.completion?(.close)
```
Comment (pixlwave) [inline ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift:57]: Would probably be sensible to switch on the action (even if there is only one right now), so that if a second is added in the future, it causes a compile error instead of unexpected behaviour.

File: ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift [hunk 2]
```diff
@@ -39,11 +46,16 @@ final class NotificationSettingsScreenCoordinator: CoordinatorProtocol {
...
+        }.store(in: &cancellables)
```
Comment (pixlwave) [inline ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift:58]:
```suggestion
        }
        .store(in: &cancellables)
```

File: ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift [hunk 3]
```diff
@@ -30,6 +35,8 @@ final class NotificationSettingsScreenCoordinator: CoordinatorProtocol {
     private let actionsSubject: PassthroughSubject<NotificationSettingsScreenCoordinatorAction, Never> = .init()
     private var cancellables: Set<AnyCancellable> = .init()
     
+    var completion: ((NotificationSettingsScreenCoordinatorResult) -> Void)?
```
Comment (pixlwave) [inline ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift:38]: We have been slowly replacing this optional `completion` with the actions publisher below. Could we use that instead please rather than adding the old one back?

File: ElementX/Sources/Screens/RoomNotificationSettingsScreen/View/RoomNotificationSettingsScreen.swift [hunk 4]
```diff
@@ -71,6 +71,9 @@ struct RoomNotificationSettingsScreen: View {
                 .compoundFormSectionHeader()
         } footer: {
             Text(context.viewState.strings.customSettingFootnote)
+                .onTapGesture {
+                    context.send(viewAction: .globalSettingsTapped)
+                }
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomNotificationSettingsScreen/View/RoomNotificationSettingsScreen.swift:76]: It would be more correct here to add a link to the string and use an OpenURLAction to send the view action instead of a custom tap gesture.

File: ElementX/Sources/Screens/RoomNotificationSettingsScreen/RoomNotificationSettingsScreenCoordinator.swift [hunk 5]
```diff
@@ -18,23 +18,51 @@ import Combine
...
+        let navigationStackCoordinator = NavigationStackCoordinator()
+        let notificationSettingsScreenParameters = NotificationSettingsScreenCoordinatorParameters(userNotificationCenter: UNUserNotificationCenter.current(),
+                                                                                                   notificationSettings: parameters.notificationSettingsProxy,
+                                                                                                   isModallyPresented: true)
+        let notificationSettingsScreenCoordinator = NotificationSettingsScreenCoordinator(parameters: notificationSettingsScreenParameters)
+        notificationSettingsScreenCoordinator.completion = { [weak self] _ in
+            self?.navigationStackCoordinator?.setSheetCoordinator(nil)
+        }
+        navigationStackCoordinator.setRootCoordinator(notificationSettingsScreenCoordinator)
+        self.navigationStackCoordinator?.setSheetCoordinator(navigationStackCoordinator)
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomNotificationSettingsScreen/RoomNotificationSettingsScreenCoordinator.swift:66]: Given this navigation as adding a sheet, it really shouldn't be done by this coordinator. The flow coordinator needs to know about any modals so that if a user tapped on a notification whilst this screen modal was presented, the flow coordinator would know the navigation state and that it needs to tear down the modal when pushing to the room.

Can you update the navigation so that this coordinator sends an action to the flow coordinator requesting the notification settings screen to be presented so it can all go through the state machine?

File: ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenModels.swift [hunk 6]
```diff
@@ -17,11 +17,10 @@
 import Foundation
 import UIKit
 
-enum NotificationSettingsScreenViewModelAction { }
-
```
Comment (pixlwave) [inline ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenModels.swift:21]: Whilst I see the logic of moving this next to the view actions, it makes the structure of this file different to all of the others where the vm actions are at the top and the view actions are at the bottom. Maybe if this is something we want to do, it should be a separate PR to update all the files consistently.

**Review by pixlwave — COMMENTED** [review]
Very nice, one thought about navigation, otherwise all the other comments are minor.

File: ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenModels.swift (hunk 6 above)
Comment (nimau) [reply ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenModels.swift]: Done.

File: ElementX/Sources/Screens/RoomNotificationSettingsScreen/RoomNotificationSettingsScreenCoordinator.swift (hunk 5 above)
Comment (nimau) [reply ElementX/Sources/Screens/RoomNotificationSettingsScreen/RoomNotificationSettingsScreenCoordinator.swift]: Oh ok, the `NotificationSettingsScreen` is now opened modally by the `RoomFlowCoordinator`

File: ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift (hunk 3 above)
Comment (nimau) [reply ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift]: done.

File: ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift (hunk 1 above)
Comment (nimau) [reply ElementX/Sources/Screens/Settings/NotificationSettingsScreen/NotificationSettingsScreenCoordinator.swift]: 👍

File: ElementX/Sources/Screens/RoomNotificationSettingsScreen/View/RoomNotificationSettingsScreen.swift (hunk 4 above)
Comment (nimau) [reply ElementX/Sources/Screens/RoomNotificationSettingsScreen/View/RoomNotificationSettingsScreen.swift:76]: On this point, I understand the logic, it's just that I'm not sure what this link should look like.

Comment (nimau) [reply ElementX/Sources/Screens/RoomNotificationSettingsScreen/View/RoomNotificationSettingsScreen.swift]: Ok, done, thanks for the help :)

File: ElementX/Sources/Screens/RoomDetailsScreen/RoomDetailsScreenCoordinator.swift [hunk 7]
```diff
@@ -29,19 +29,24 @@ struct RoomDetailsScreenCoordinatorParameters {
...
     private var navigationStackCoordinator: NavigationStackCoordinator? {
         parameters.navigationStackCoordinator
     }
     
-    var callback: ((RoomDetailsScreenCoordinatorAction) -> Void)?
-    
+    private let actionsSubject: PassthroughSubject<RoomDetailsScreenCoordinatorAction, Never> = .init()
+    private var cancellables: Set<AnyCancellable> = .init()
+        
+    var actions: AnyPublisher<RoomDetailsScreenCoordinatorAction, Never> {
```
Comment (pixlwave) [inline ElementX/Sources/Screens/RoomDetailsScreen/RoomDetailsScreenCoordinator.swift:46]: Oh nice, thanks for adding it here too 👍

**Review by pixlwave — APPROVED** [review]
Perfect, looks great to me! 😎
